hoomd/md/update.py

```diff
@@ -221,100 +224,140 @@ class mueller_plathe_flow(_updater):
+        # const integrated flow with 0.1 slope for max 1e8 timesteps
+        ramp = hoomd.variant.Ramp(0.0, 0.1e8, 0, int(1e8))
+        # velocity gradient in z direction and shear flow in x direction.
+        slab_direction = hoomd.md.update.MullerPlatheFlow.Z
+        flow_direction = hoomd.md.update.MullerPlatheFlow.X
+        mpf = hoomd.md.update.MuellerPlatheFlow(filter = hoomd.filter.All(), flow_target = ramp, slab_direction  = slab_direction, flow_direction = flow_direction, n_slabs = 20)
 
-        self.setupUpdater(period);
 
-    def get_n_slabs(self):
-        R""" Get the number of slabs."""
-        return self.cpp_updater.getNSlabs()
+    Attributes:
+        filter (hoomd.filter.ParticleFilter): Subset of particles on which to
+            apply this updater.
 
-    def get_min_slab(self):
-        R""" Get the slab id of min velocity search."""
-        return self.cpp_updater.getMinSlab()
+        flow_target (hoomd.variant.Variant): Integrated target flow in the
+            natural units of the simulation.
 
-    def get_max_slab(self):
-        R""" Get the slab id of max velocity search."""
-        return self.cpp_updater.getMaxSlab()
+        slab_direction (``X``, ``Y``, or ``Z``): Direction perpendicular to the
+            slabs.
 
-    def get_flow_epsilon(self):
-        R""" Get the tolerance between target flow and actual achieved flow."""
-        return self.cpp_updater.getFlowEpsilon()
+        flow_direction (``X``, ``Y``, or ``Z``): Direction of the flow.
 
-    def set_flow_epsilon(self,epsilon):
-        R""" Set the tolerance between target flow and actual achieved flow.
+        n_slabs (int): Number of slabs.
 
-           Args:
-           epsilon (float): New tolerance for the deviation of actual and achieved flow.
+        max_slab (int): Id < n_slabs where the max velocity component is
+            searched for.
 
-        """
-        return self.cpp_updater.setFlowEpsilon(float(epsilon))
+        min_slab (int): Id < n_slabs where the min velocity component is
+            searched for.
+    """
+    def __init__(self, filter, flow_target, slab_direction, flow_direction ,n_slabs, max_slab=-1, min_slab=-1):
 
-    def get_summed_exchanged_momentum(self):
+        assert (n_slabs > 0), "Invalid negative number of slabs."
+        if min_slab < 0:
+            min_slab = 0
+        if max_slab < 0:
+            max_slab = n_slabs / 2
+        assert (max_slab > -1 and max_slab < n_slabs), "Invalid max_slab in [0,"+str(n_slabs)+")."
+        assert (min_slab > -1 and min_slab < n_slabs), "Invalid min_slab in [0,"+str(n_slabs)+")."
+        assert (min_slab != max_slab), "Invalid min/max slabs. Both have the same value."
+
+        params = ParameterDict(
+            filter=hoomd.filter.ParticleFilter,
+            flow_target=hoomd.variant.Variant,
+            slab_direction=_md.MuellerPlatheFlow.Direction,
```

Comment:
Let's encode the directions as strings at the Python level. Pass them as std::string to C++ methods which can set the underlying enum values appropriately.